lib/web/fetch/decompress.js

```diff
@@ -0,0 +1,95 @@
+const { pipeline } = require('node:stream')
+const zlib = require('node:zlib')
+const { redirectStatusSet, nullBodyStatus } = require('./constants')
+const { createInflate } = require('./util')
+
+/**
+ * Decompress the given response body.
+ * @param {Request} request
+ * @param {Response} response
+ * @returns {ReadableStream<Uint8Array> | null}
+ */
+function decompress (request, response) {
+  if (response.body === null) {
+    return response.body
+  }
+
+  const willFollow =
+    response.headers.get('location') &&
+    request.redirect === 'follow' &&
+    redirectStatusSet.has(response.status)
+
+  if (
+    request.method === 'HEAD' ||
+    request.method === 'CONNECT' ||
+    nullBodyStatus.includes(response.status) ||
+    willFollow
+  ) {
+    return response.body
+  }
+
+  return decompressStream(
+    response.body,
+    response.headers.get('content-encoding')
+  )
+}
+
+/**
+ * Decompress the given stream based on the "Content-Encoding" response header.
+ * @param {ReadableStream<Uint8Array>} stream
+ * @param {string | null} contentEncoding The value of the "Content-Encoding" response header.
+ */
+function decompressStream (stream, contentEncoding) {
```

Comment:
Wondering if it might be useful to allow customize the options for each one of the decompressions implementation.

Reply:
@metcoder95, do we have a use case for that? I'd love to see it, think how we can support additional parameters. Perhaps the way forward would be through more granular utilities, if any, then making one function with complex input.

Reply:
Hmm, that's a good point; let's leave it for later then. I don't have a specific use-case in mind but just if you want to add extra params to e.g. `brotli` decompress, or strategy to `gunzip`.

But unless strong request, we can keep it simple as you suggest 👍